services/fileuploader/fileuploader.go

```diff
@@ -173,8 +173,8 @@ func overrideWithSettings(defaultConfig map[string]interface{}, settings backend
 	for k, v := range settings.Config {
 		config[k] = v
 	}
-	if _, ok := config["externalId"]; ok && settings.Type == "S3" {
-		config["externalId"] = workspaceID
+	if settings.Type == "S3" {
+		config["externalID"] = workspaceID
```

Comment:
is it `externalID` or `externalId`?

Reply:
The reference that I've gone with is what we do [here](https://github.com/rudderlabs/rudder-server/blob/master/utils/misc/misc.go#L931)
same for the above comment as well, just went ahead without checking for `iamRoleArn` because it's not being done in above reference.
I wasn't sure of what's required, but was sure that the batchrouter succeeds, so simply just went ahead.

Reply:
it is confusing that there are occurrences where we are using `externalID` and others where we are using `externalId`... Are both correct in their own context, or something needs fixing? @koladilip

Reply:
`externalID` is used in session util which is the one that generates role credentials using externalID. Not sure if externalId works